flatland_server/include/flatland_server/message_server.h

```diff
@@ -0,0 +1,244 @@
+#ifndef FLATLAND_SERVER_MESSAGE_SERVER_H
+#define FLATLAND_SERVER_MESSAGE_SERVER_H
+
+#include <deque>
+#include <unordered_map>
+#include <vector>
+
+#include <ros/ros.h>
+
+namespace flatland_server {
+
+class MessageServer;
+class MessageTopicBase {
+ public:
+  virtual void clean_old_messages() = 0;
+};
+class PublisherBase {};
+class SubscriberBase {};
+
+template <class T>
+class MessageView {
+  typename std::deque<std::pair<ros::Time, T>>::const_iterator begin_;
```

Comment:
You can clean this up a little with a using directive.
e.g., `using ConstIterator = std::deque<std::pair<ros::Time, T>>::const_iterator`